### PR TITLE
Make Decode canonicalize CBOR.

### PR DESCRIPTION
**Pull request by Stebalien**
Also:

* Actually use pre-computed CIDs (trivial bug fix).
* Cleanup DecodeInto and make it possible to DecodeInto non-objects. This came up when refactoring the decoding logic.
* `gx import` go-block-format because I apparently forgot to do this.
* `gx release 1.2.4` so we can use this in [ipfs/go-ipfs/#4023](https://github.com/ipfs/go-ipfs/issues/4023)

I've also imported a variant of the canonicalization test so I don't re-introduce this bug in the future...

### CODE REVIEW

File: node_test.go [hunk 1]
```diff
@@ -333,3 +368,51 @@ func TestExamples(t *testing.T) {
...
+	}
+	nd2, err := Decode(nd1.RawData())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !nd2.Cid().Equals(nd1.Cid()) || !bytes.Equal(nd2.RawData(), nd1.RawData()) {
+		t.Fatal("re-decoding a canonical node should be idempotent")
+	}
+}
```
Comment (magik6k) [inline node_test.go]: If test_objects/non-canon.cbor is taken from go-ipfs I'd add a check for whether CID is `zdpuAmxF8q6iTUtkB3xtEYzmc5Sw762qwQJftt5iW8NTWLtjC` here

File: node.go [hunk 2]
```diff
@@ -77,6 +63,19 @@ func DecodeInto(b []byte, v interface{}) error {
 
 }
 
+// Decode's a cbor node into an object.
```
Comment (magik6k) [inline node.go]: `Decodes`

File: node_test.go (hunk 1 above)
Comment (Stebalien) [inline node_test.go]: Done.

File: node.go (hunk 2 above)
Comment (Stebalien) [inline node.go]: :walking_man: :notes:

**Comment by Stebalien** [comment]
@whyrusleeping it should all be fixed (I've also added a bit of documentation to the Decode/DecodeBlock functions).

In the future, do you prefer keeping history intact while a PR is under review?

**Comment by magik6k** [comment]
Updated https://github.com/ipfs/go-ipfs/pull/4023 with current gx version of this to see if tests pass

**Pull request merged**